### PR TITLE
Add download tag for self contained html

### LINKED ISSUE #273: Support FORMAT_TEXT on recent chrome with self contained html
Right now doing something like this:

```
import pytest

from pytest_html import extras

def test_ex(extra):
    print ("hey")
    extra.append(extras.text("Hello World!\n", "Hello Text"))
```

leads to something like this:

image

The problem is if you click the "Hello Text" link, chrome will close it immediately. This is a standard behavior as of recent chrome (https://stackoverflow.com/questions/46818033/opening-datatext-plain-window-in-chrome-showing-up-blank). 

Can we come up with a different way to make the text openable/viewable in self contained html?

### PR DESCRIPTION
**Pull request by csm10495**
This should resolve #273. If we're working with a self contained html, simply download text (and json) files instead of trying to open them in the current browser window.

### CODE REVIEW

**Comment by csm10495** [comment]
If we think the logic makes sense, I can work to try to fix the unit tests, otherwise, I'll leave this as is for now for further thoughts.

**Comment by BeyondEvil** [comment]
This was to deal with something that Chrome (and IE/Edge) did (block data content).

I wonder if we only should change the behavior for those browsers (which would require us to deal with this in javascript) or if we sort of agree with the security aspect and change it for all the things?

I guess what I'm a little bit worried about is introducing a breaking change. 🤔

**Comment by csm10495** [comment]
I think Firefox also blocks it...
https://blog.mozilla.org/security/2017/11/27/blocking-top-level-navigations-data-urls-firefox-59/

File: pytest_html/plugin.py [hunk 1]
```diff
@@ -254,6 +258,7 @@ def append_extra_html(self, extra, extra_index, test_index):
                         class_=extra.get("format"),
                         href=href,
                         target="_blank",
+                        download=download,
```
Comment (christiansandberg) [inline pytest_html/plugin.py]: For non self-contained HTML this becomes `download="None"`. You probably need to add the argument conditionally somehow.

File: pytest_html/plugin.py [hunk 2]
```diff
@@ -236,6 +239,7 @@ def append_extra_html(self, extra, extra_index, test_index):
                     content = content.decode("utf-8")
                 if self.self_contained:
                     href = data_uri(content)
+                    download = extra.get("name")
```
Comment (christiansandberg) [inline pytest_html/plugin.py]: You should add extension to the filename. Browsers seem to figure it out based on mime type but it is probably best to be explicit.

File: pytest_html/plugin.py (hunk 1 above)
Comment (BeyondEvil) [reply pytest_html/plugin.py]:
```python
kwargs = {
    "class_": extra.get("format"),
    "href": href,
    "target": "_blank",
}

if download:
    kwargs["download"] = download

href = self.create_asset(
    content, extra_index, test_index, extra.get("extension"), **kwargs,
)
```

File: pytest_html/plugin.py (hunk 2 above)
Comment (csm10495) [reply pytest_html/plugin.py]: Done (I think)

File: pytest_html/plugin.py (hunk 1 above)
Comment (csm10495) [reply pytest_html/plugin.py]: Done (I think)

**Comment by csm10495** [comment]
Added a couple tests and (hopefully) fixed the linting. @BeyondEvil give it one more look if you can. Thanks.

**Comment by csm10495** [comment]
@ssbarnea any chance this or could get merged (after rebase)?

**Comment by ssbarnea** [comment]
After rebase yes but it would also be very useful to provide a link in PR description to an sample file produced with this feature, so we can easily test how it works.

**Comment by csm10495** [comment]
This fell wayyy off my radar. Closing out since i haven't been able to get to it. Sorry folks.

**Pull request closed without merging**